server/src/main/java/org/opensearch/index/IndexModule.java

```diff
@@ -604,8 +605,8 @@ public IndexService newIndexService(
         IndexStorePlugin.DirectoryFactory remoteDirectoryFactory,
         BiFunction<IndexSettings, ShardRouting, TranslogFactory> translogFactorySupplier,
         Supplier<TimeValue> clusterDefaultRefreshIntervalSupplier,
```

Comment:
Possible to create a wrapper class for the cluster level settings as well?

Reply:
I think so. Will create an issue to discuss this further.